### PR TITLE
PYTHON-4287: Support Creation of Multiple Vector Search Indexes for Database Generation Step

### PR DESCRIPTION
**Pull request by Jibola**
Support multiple index creation for our AI-ML testing pipeline

### CODE REVIEW

**Review by caseyclements — CHANGES_REQUESTED** [review]
There are no tests for this feature. We need tests of the testing framework itself.

File: .evergreen/provision-atlas.sh [hunk 1]
```diff
@@ -42,6 +42,8 @@ DATABASE=$DATABASE \
     $PYTHON_BINARY $SCAFFOLD_SCRIPT
 
 # If a search index configuration can be found, create the index
-if [ -f "$TARGET_DIR/indexConfig.json" ]; then
-    $atlas deployments search indexes create --file $TARGET_DIR/indexConfig.json --deploymentName $DIR
+if [ -f "$TARGET_DIR/indexes" ]; then
```
Comment (Jibola) [inline .evergreen/provision-atlas.sh:45]:
```suggestion
if [ -d "$TARGET_DIR/indexes" ]; then
```

**Comment by Jibola** [comment]
> There are no tests for this feature. We need tests of the testing framework itself.

I agree here. I'm thinking we should make this a separate task seeing as there's quite a bit that would need tests.

File: README.md [hunk 2]
```diff
@@ -19,23 +20,26 @@ Each AI/ML pipeline is sorted by the composite of the name of the library, and t
...
 Within each subdirectory you should expect to have:
-- `run.sh`  --  A script that should handle any additional library installations and steps for executing the test suite. This script should not populate the Atlas database with any required test data.
+
+- `run.sh` -- A script that should handle any additional library installations and steps for executing the test suite. This script should not populate the Atlas database with any required test data.
 - `database/` -- An optional directory used by `.evergreen/scaffold_atlas.py` to populate a MongoDB database with test data. Only provide this if your tests require pre-populated data.
 - `database/{collection}.json` -- An optional JSON file containing one or more MongoDB documents that will be uploaded to `$DATABASE.{collection}` in the local Atlas instance. Only provide this if your tests require pre-populated data.
 - `indexConfig.json` -- An optional file containing configuration for a specified Atlas Search Index.
 - Additionally, you can add other useful files, including `.env` files, if required by your tests.
 
 The general layout of this repo will looks like this:
```
Comment (caseyclements) [inline README.md:30]: The general layout of this repo **will look** like this:

OR

The general layout of this repo **looks** like this:

**Review by caseyclements — CHANGES_REQUESTED** [review]
I'm keen to get this one merged, as docarray needs this.

File: README.md (hunk 2 above)
Comment (Jibola) [reply README.md:30]: > The general layout of this repo looks like this:

File: .evergreen/provision-atlas.sh [hunk 3]
```diff
@@ -42,6 +42,8 @@ DATABASE=$DATABASE \
     $PYTHON_BINARY $SCAFFOLD_SCRIPT
 
 # If a search index configuration can be found, create the index
-if [ -f "$TARGET_DIR/indexConfig.json" ]; then
-    $atlas deployments search indexes create --file $TARGET_DIR/indexConfig.json --deploymentName $DIR
```
[the remark this answers is not in the document]
Comment (Jibola) [reply .evergreen/provision-atlas.sh:46]: Yep. I just sync'd the branch with main and made the updates to index hierarchy

**Review by caseyclements — APPROVED** [review]
LGTM!

**Pull request merged**